src/Admin.php

```diff
@@ -7,110 +7,160 @@
+                  'type' => 'select',
+                  'choices' => self::get_shipping_classes(),
+                  'default_value' => [],
+                  'allow_null' => 0,
+                  'multiple' => 1,
+                  'ui' => 1,
+                  'return_format' => 'value',
+                ],
+                [
+                  'key' => 'shipping_class_inner_rules',
+                  'label' => __('Shipping Class Inner Rules', Plugin::L10N),
+                  'name' => 'shipping_class_inner_rules',
+                  'type' => 'repeater',
+                  'layout' => 'block',
+                  'button_label' => 'Add rule',
+                  'sub_fields' => [
+                    [
+                      'key' => 'shipping_info',
+                      'label' => __('Shipping info text', Plugin::L10N),
+                      'name' => 'shipping_info',
+                      'type' => 'text',
+                      'required' => 1,
+                    ],
+                    [
+                      'key' => 'min_price',
+                      'label' => __('Min price', Plugin::L10N),
+                      'name' => 'min_price',
+                      'type' => 'number',
+                      'required' => 1,
+                    ],
+                    [
+                      'key' => 'country',
+                      'label' => __('Countries', Plugin::L10N),
+                      'name' => 'country',
+                      'type' => 'select',
+                      'choices' => self::get_shipping_countries(),
+                      'default_value' => [],
+                      'allow_null' => 0,
+                      'multiple' => 1,
+                      'required' => 1,
+                      'ui' => 1,
+                      'return_format' => 'value',
+                    ],
+                  ],
+                ],
+              ],
+            ],
+          ],
+          'location' => [
+            [
+              [
+                'param' => 'options_page',
+                'operator' => '==',
+                'value' => 'dynamic-shipping-info',
+              ],
+            ],
+          ],
+        ]
+      );
+
+    endif;
   }
 
   /**
-   * Adds settings fields to corresponding WooCommerce settings section.
+   * Gets dynamic shipping info rules ordered by shipping class name.
    *
-   * @implements woocommerce_settings_<current_tab>
+   * @return array
+   *   Array of defined dynamic shipping info rules ordered by shipping class name.
    */
-  public static function woocommerce_settings_dynamic_shipping_info() {
-    $settings = static::getSettings();
-    WC_Admin_Settings::output_fields($settings);
+  public static function get_dynamic_shipping_rules() {
+    $dynamic_shipping_info_rule = get_field('dynamic_shipping_info_rule', 'option') ?: [];
+    usort($dynamic_shipping_info_rule, function ($a, $b) {
+      return $b['shipping_class'] <=> $a['shipping_class'];
```

Comment:
In case you wonder why this is needed, is because I want to put the rules with empty shipping class at the end, avoiding thus overlapping.